### ISSUE TITLE
On rotation to landscape, no longer see the new / old color boxes

### ISSUE THREAD

**Issue by mikeyscott** [danielnilsson9/color-picker-view#2]
On both the devices I used this on within my own app (as a Preference for color selection) it worked great in portrait mode, but whenever I switch to landscape, the saturation color selector square eats up the entire height of the screen and leaves no room for the new/old color boxes (see screen shot), rendering the dialog unusable. One device was a Moto G phone and the other was a Samsung Galaxy Tab 4 (10.1"), both running Android 4.4.2.

![screenshot_2015-05-07-11-44-18](https://cloud.githubusercontent.com/assets/7632058/7519566/ff6bb870-f4ae-11e4-9348-b2b6e12a45d9.png)

**Comment by danielnilsson9** [comment]
Im working on it, the ColorPickerDialog will be replaced by a ColorPickerDialogFragment instead since those AlertDialogs seems to be hard to control... Did not have that on any of my devices, very strange.

**Comment by mikeyscott** [comment]
Great.. thank you.

**Comment by danielnilsson9** [comment]
Should be ok now, please try the new library.

I have tested on:
- LG G3 Android 5.0
- HTC Nexus 9 Android 5.0.2
- ASUS Transformer TF101 Android 4.4.4
- HTC One X Android 4.2.2
- Sony Xperia Neo Android 4.0.4
- HTC Sensation Android 4.0.3

**Comment by mikeyscott** [comment]
Looks great thank you. Modified it slightly to work with my preference fragment (not activity) but I like the revised layout & explicit Ok button, and works well on all three of my test devices both landscape & portrait. 

![screenshot_2015-05-09-19-25-40](https://cloud.githubusercontent.com/assets/7632058/7552435/e368ffc8-f681-11e4-93c5-21c4de7f6ffc.png)

**Comment by danielnilsson9** [comment]
Looks good! I guess I was a bit quick in the last testing since it did look very good on sw320dp displays.
I have fixed that now along with a bunch of other issues. One if them we can see on your screenshot, the seleced hue value doesn't really match what is shown in the sat/val panel, yellow is seleced, green is shown.

Also the demo app is updated with an example on how to use the ColorPreference from a PreferenceFragment.

**Comment by mikeyscott** [comment]
All looks good now apart from one last issue I encountered which was if the initial color was white (-1) then the current / new color panels showed up as black rather than white. Seemed like this bit of code in ColorPickerDialogFragment.onCreateView was filtering out the 'white' case but I dont know why?

```
        int initColor = getArguments().getInt("init_color");

        if(initColor != -1) {
            mOldColorPanel.setColor(initColor);
            mColorPicker.setColor(initColor, true);
        }
```

Once I commented out the test of initColor != -1 it all worked fine.
